### ISSUE TITLE
Full Resolution Photo and Video Sharing Option

### ISSUE THREAD

**Issue by pmr0** [mollyim/mollyim-android#277]
### Is there an existing request for this?

- [X] I have searched the existing issues

### Feature description

Currently, when sharing photos and videos through the Molly (and Signal), they undergo resizing or compression, which often leads to a loss in quality. In addition to the Standard and High quality options, an Original option could be included.

My suggestion is to introduce an option allowing users to send photos and videos in their original, full resolution without any alterations. This feature would be particularly valuable for individuals who wish to preserve the high quality and clarity of their visual content when sharing.

Signal's reason for not allowing full resolution images is not quite convincing. Other IM apps allow sending of full resolution photos and videos (with a size limit of course for very large files).

**Comment by kasia125** [comment]
I came to jokingly suggest this because there is less code executed when not invoking the media compression libraries.

**Comment by kasia125** [comment]
>     1. I saw some signal groups forums where people were saying that sending a `File` not Media will send video/picture without quality loss. Is it true or not?

I think that sending media as a file might cause other clients to not view it as a view, so the people will need to download the file instead.

>     2. I understand that sending smaller file is more convenient in most of the cases and I agree with the default option we have now. But the lack of ability to send full size video/picture makes me think the video goes to some sort of server, so the trim or quality loss is required to keep server not too busy. If it's true then it is not P2P communication. But if it's truly P2P, the option to send original should not be an issue.

No, it's not P2P... your files go to an AWS server, and Signal stores them for up to 45 days. The files should be encrypted, but everything goes through a server and is not P2P.
Smaller files means:
- you upload less data
- your recipients download less data
- Signal's server stores less data

The only communication that is P2P are voice calls, where possible, and where both parties have set Signal to not relay the call through the server.

>     3. Sometimes I would like to reduce the video quality, but to keep the video in full length without trimming. Currently I am forced to trim the video, again I don't understand why?

Are you doing this manually, or do you not understand the issue? You should only need to do this manually if your videos are over somewhere around 90MiB.

>     4. Could agree with such limitations for groups only.

Why for groups only? The file is uploaded only once, then the key for the file is shared to every recipient, so they can download then decrypt the file.

**Comment by nisc** [comment]
I'd also appreciate this feature (more like a nice-to-have, given there are already workarounds today), but this sounds like something that would best be done upstream, given Signal servers are involved and it could lead to quirks in the official client?

I think upstream lobbying is the better approach.

**Comment by fkdp06** [comment]
> No, it's not P2P... your files go to an AWS server, and Signal stores them for up to 45 days.

The above issue is very obvious for me, and I don't understand why it's not done the following way. When you establishing a call you stream the data (voice) straight to the recipient (P2P) am I correct? If it's true, why then the app cannot stream the original full size file straight to that recipient omitting the server?

Otherwise I don't see why Signal is different from all other messaging apps like Whatsapp or Telegram? Originally I thought the server is required only to find the contact's ip route in order to establish the connection using some kind of [DHT](https://en.wikipedia.org/wiki/Distributed_hash_table).

**Comment by kasia-de** [comment]
> I'd also appreciate this feature (more like a nice-to-have, given there are already workarounds today), but this sounds like something that would best be done upstream, given Signal servers are involved and it could lead to quirks in the official client?

It's really only a client-side-only thing. We can confirm by looking at the source from when compression was introduced

**Comment by kasia-de** [comment]
> The above issue is very obvious for me, and I don't understand why it's not done the following way. When you establishing a call you stream the data (voice) straight to the recipient (P2P) am I correct?

You are correct

> If it's true, why then the app cannot stream the original full size file straight to that recipient omitting the server?

Well,
1. Just like calling, you can't do this if the recipient is offline! Making this a requirement would disappoint a lot of people, but it's a common pain point in P2P messaging.
2. If you send a 45MiB file to a group chat with 11 people, your device now needs to upload 450MiB (45 * (11 - 1)), to delipver 45MiB to each of the people. By uploading to the server, you are offloading some of the work.
3. Signal simply isn't designed this way. I know, it's unfortunate, although there _are_ other messagers out there! Maybe you would be interested in [Cwtch](https://cwtch.im) or [Quiet](https://github.com/TryQuiet/quiet) instead of Signal?
